src/lib/transaction_logic/test/account_timing.ml

```diff
@@ -0,0 +1,194 @@
+open Core_kernel
+open Mina_transaction_logic
+open Currency
+open Mina_base
+open Mina_numbers
+
+type timing = (Global_slot.t, Balance.t, Amount.t) Account_timing.tt
+[@@deriving compare, sexp]
+
+type txn_result = (timing, Error.t) Result.t [@@deriving compare, sexp]
+
+let final_vesting_slot ~initial_minimum_balance ~cliff_time ~cliff_amount
+    ~vesting_period ~vesting_increment =
+  let open Unsigned in
+  let to_vest =
+    Balance.(initial_minimum_balance - cliff_amount)
+    |> Option.value_map ~default:UInt64.zero ~f:Balance.to_uint64
+  in
+  let incr = Amount.to_uint64 vesting_increment in
+  let periods =
+    let open UInt64 in
+    if compare (rem to_vest incr) zero = 0 then div to_vest incr
+    else succ @@ div to_vest incr
+  in
+  let open UInt32 in
+  let open Infix in
+  Global_slot.to_uint32 cliff_time
+  + (UInt64.to_uint32 periods * Global_slot.to_uint32 vesting_period)
+  |> Global_slot.of_uint32
+
+let timing_final_vesting_slot = function
+  | Account.Timing.Untimed ->
+      Global_slot.zero
```

Comment:
again, using `zero` to encode `None` is probably not a good strategy

Reply:
These are account timing tests and the function is only use locally, always on timed accounts. Actually, the necessity to unpack options here would be quite annoying, considering that generators guarantee that generated accounts will be properly timed. I think the risk of someone calling this function on an untimed account by accident is very low. And even if that happens, the test will likely be failing for that reason so the CI will prevent merging it.